pkg/resources/tag_acceptance_test.go

```diff
@@ -38,52 +57,341 @@ func TestAcc_Tag(t *testing.T) {
 		},
 		CheckDestroy: acc.CheckDestroy(t, resources.Tag),
 		Steps: []resource.TestStep{
+			// base model
 			{
-				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_Tag/basic"),
-				ConfigVariables: m(),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(resourceName, "name", id.Name()),
-					resource.TestCheckResourceAttr(resourceName, "fully_qualified_name", id.FullyQualifiedName()),
-					resource.TestCheckResourceAttr(resourceName, "database", acc.TestDatabaseName),
-					resource.TestCheckResourceAttr(resourceName, "schema", acc.TestSchemaName),
-					resource.TestCheckResourceAttr(resourceName, "allowed_values.#", "1"),
-					resource.TestCheckResourceAttr(resourceName, "allowed_values.0", ""),
-					resource.TestCheckResourceAttr(resourceName, "comment", "Terraform acceptance test"),
+				Config: config.FromModel(t, baseModel),
+				Check: assert.AssertThat(t, resourceassert.TagResource(t, baseModel.ResourceReference()).
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()).
+					HasCommentString("").
+					HasNoMaskingPolicy(),
```

Comment:
nit: there can be multiple masking policies, right? So why we have a singular name?

Reply:
Fixed.